Source/Initialization/WarpXInitData.cpp

```diff
@@ -211,6 +213,69 @@ WarpX::ComputePMLFactors ()
     }
 }
 
+void
+WarpX::ComputeMaxStep ()
+{
+    if (do_compute_max_step_from_zmax) {
+        computeMaxStepBoostAccelerator(geom[0]);
+    }
+
+    // Make max_step consistent with stop_time, assuming constant dt. The if condition is here to
+    // avoid an overflow in the static_cast.
+    if (istep[0] + (std::ceil( (stop_time-t_new[0])/dt[0] )) < std::numeric_limits<int>::max()) {
```

Comment:
With the current logic, there are a few risks with the overflow handling (the overlowing number becoming negative, which is smaller than max, then settling somewhere in the value space).

I am thinking to maybe change it like this:
- calculate `max_step * dt`
- compare `max_step * dt` & `stop_time`, find the smaller (in seconds)
- use the "step" of the smaller one for `max_step`

Reply:
Can floats become negative on overflow? I was expecting that the value would become `inf` in this case, which should not be smaller than `std::numeric_limits<int>::max()`, but maybe I'm mistaken here.

Reply:
Yes, both can overfloat, become Inf, NaN and negative. But practically speaking, I would not expect that we quickly throw the float range here (we just loose precision but the exponents are pretty large).

If an int overflows, this results in undefined behavior. This often is implemented to wrap around to negative numbers (but can be anything).

A [float overflow](https://stackoverflow.com/questions/17588419/when-a-float-variable-goes-out-of-the-float-limits-what-happens) (probably also technically undefined) usually results in an `Inf`. An `Inf` value will [be larger than](https://stackoverflow.com/questions/52817510/compare-infinity-with-infinity-in-c) `max`.

Reply:
Ok, thanks.

So if we can ignore possible float overflows, I guess the way it was done was already safe (maybe `istep[0]` should have been included in the static cast, in case the step number is somehow negative).

In any case, I liked the idea of doing the comparison in physical time rather than number of timesteps so I have implemented something closer to what you suggested.